app/models/katello/glue/pulp/repo.rb

```diff
@@ -361,16 +361,12 @@ def full_path(smart_proxy = nil, force_https = false)
       scheme   = (self.unprotected && !force_https) ? 'http' : 'https'
```

Comment:
Sorry one last comment :)

With pulp2, unprotected meant that it was only available via http (none were available via https without a special cert).  

But with pulp3, unprotected means that a user can browse the repo over https without a special cert.  Because of this, i think we should default to https on here (instead of defaulting to http).   This way a user browsing the repo details will get an https:// url by default. 

The only time this needs to return http i think is here: app/services/katello/managed_content_medium_provider.rb:12:      url = kickstart_repo.full_path(entity.content_source)

as its returning a kickstart url, it generally needs to be http://

Reply:
If the default is https, then under what conditions would we set to http? Just kickstart urls?

Reply:
yes, i don't think theres any reason to return an http:// url otherwise.  the user could always try it via http (and maybe we should show both in the UI if both are available).  that might be another issue for another day though